### DISCUSSION TITLE
Windows: How to install plugins?

### DISCUSSION THREAD

**Discussion by AviationRunning** [xournalpp/xournalpp#5279] [Q&A]
As the title suggests, I'd like to learn how to install plugins! I am completely new to this so I'll need a step by step (if you can provide!)
I am on windows 11, and there is a few things I'd like to learn how to install. I also would love to know if there is some plugins for a few things I am looking for, because searching through the discussions, I don't exactly understand the terminology or how things work.

The first thing I'd like to install is [Fly-Pie](https://github.com/Schneegans/Fly-Pie)
I also saw Mr. Rolando talk about this [plugin](https://github.com/xournalpp/xournalpp/discussions/4658). This seems to be as close as we can get to different types of pens, at the moment? 
Would I just create a new text file in the plugins folder? I'm not sure.

Another thing I would like to know is if there is a plugin to add sticky notes or mind maps. I was using GoodNotes before this and was getting sick of how restricted it was. But it did have these handy little things that helped organize my notes a little better. The only thing that I saw that MIGHT be similar is this [Cairo Library](https://www.cairographics.org/) maybe? I am not entirely sure. Attached is the pictures of the mind-maps and sticky notes. Just images you can add to your page.
![image](https://github.com/xournalpp/xournalpp/assets/76266337/0227d0c9-3691-4327-aa15-c3fc038daa26)
![image](https://github.com/xournalpp/xournalpp/assets/76266337/2423ccb1-8f11-4a37-8f06-63729216a108)

**Comment by rolandlo** [comment]
About Fly-Pie: This is only a Gnome shell extension (i.e. for part of Linux). It can't be used on Windows.

In order to write a plugin, you need to create a `plugin.ini` file (that contains some general info about the plugin) and a `main.lua` file (that contains the Lua code). See https://xournalpp.github.io/guide/plugins/plugins/#plugin-structure
For the `main.lua` file you can basically just copy the Lua code you saw in https://github.com/xournalpp/xournalpp/discussions/4658
For the `plugin.ini` file you can copy [this file](https://xournalpp.github.io/guide/plugins/plugins/#pluginini) and modify the author name and description.

On Windows the `plugin.ini` and `main.lua` file can be put either into `C:\Programs\Xournal++\share\xournalpp\plugins\PLUGINNAME\`, where you replace `PLUGINNAME` by the name you want to give to the plugin.
Alternatively (recommended) you can also put them into `C:\Users\<user>\AppData\Local\xournalpp\plugins\PLUGINNAME`, where `<user>` is your user name. You may need to create the plugins subfolder of xournalpp in that case.

Writing a plugin to add "sticky notes" to a Xournal++ document is a bit tricky, since all that makes a Xournal++ document are the background, strokes, images, latex images and text. Your plugin could create some filled strokes looking like the shapes above (and some accompanying text if you wish). Would that suit your needs?

**Reply by AviationRunning** [reply]
This was extremely helpful, I couldn't find that guide on the plugins. How unfortunate that windows cant use Gnome, guess I'll have to switch to Linux or duel boot. The sticky notes idea you had would work great, honestly! Exactly what I was going for.

**Reply by mutasim-ali-med** [reply]
> This was extremely helpful, I couldn't find that guide on the plugins. How unfortunate that windows cant use Gnome, guess I'll have to switch to Linux or duel boot. The sticky notes idea you had would work great, honestly! Exactly what I was going for.

if you made the sticky note plugin, please share to me cause i don't know how to code
